src/jrd/recsrc/IndexTableScan.cpp

```diff
@@ -252,6 +255,14 @@ bool IndexTableScan::internalGetRecord(thread_db* tdbb) const
 				continue;
 			}
 
+			// If we're walking in a descending index and we need to ignore NULLs
+			// then stop at the first NULL we see (only for single segment!)
+			if (descending && ignoreNulls && node.prefix == 0 &&
+				node.length >= 1 && node.data[0] == 255)
```

Comment:
Accordingly to the `compress()`: 
// Further a NULL state is always returned as 1 byte 0xFF (descending index).
So, why check for `>= 1`, not `== 1` ? 
And, for consistency, please use 0xFF, not 255.

Reply:
This is a copy-paste from btr.cpp :) where such a check is used twice. I don't mind changing it, but I'd suggest this to be done in btr.cpp too.

Reply:
I'm also wondering whether we really should check for `(node.prefix == 0 && node.length == 1 && node.data[0] == 0xFF)` -- i.e. the first NULL encountered -- or better for `(node.prefix + node.length == 1 && node.data[0] == 0xFF)` -- i.e. any NULL encountered, as more reliable. Is it theoretically possible that we somehow jump to the bucket in the middle of the NULL duplicates chain and start scanning from there, thus skipping the first NULL node?

Reply:
> I'm also wondering whether we really should check for 
`(node.prefix == 0 && node.length == 1 && node.data[0] == 0xFF)` -- i.e. the first NULL encountered -- or better 
for 
`(node.prefix + node.length == 1 && node.data[0] == 0xFF)` -- i.e. any NULL encountered, as more reliable. 

Second way is not correct: when `node.length == 0` we should not access `node.data[0]` at all.
Instead, we could look at `key` contents, if necessary.

> Is it theoretically possible that we somehow jump to the bucket in the middle of the NULL duplicates chain and start scanning from there, thus skipping the first NULL node?

The position is based on the last key processed, so, we should be safe.

Reply:
> This is a copy-paste from btr.cpp :) where such a check is used twice. I don't mind changing it, but I'd suggest this to be done in btr.cpp too.

Ok, then, let it be this way